python/rapidsmpf/rapidsmpf/shuffler.pyx

```diff
@@ -131,6 +133,9 @@ cdef class Shuffler:
         chunks
             A map where keys are partition IDs (``int``) and values are packed
             data (``PackedData``).
+        grouped
+            If ``True``, the chunks are grouped by the destination rank of the
+            partition ID.
```

Comment:
A bit of context around when a user might (or might not) want to do this. IIUC, it's most beneficial when there are many partitions relative to the number of nodes because fewer messages are required. Is there any downside to `grouped=True`?